lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,26 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @warriors = building.warriors
+      @granary = building.granary
     end
 
     def call
-      raise NotImprementedYet
+      @warriors.present? ? @granary / daily_consumption : 0
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś wyżej, żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)